src/UnitfulRecipes.jl

```diff
@@ -124,6 +125,25 @@ end
 #===============
 Attribute fixing
 ===============#
+# Aspect ratio
+function fixaspectratio!(attr, u, axisletter)
+    aspect_ratio = get!(attr, :aspect_ratio, :auto)
+    if aspect_ratio in (:auto, :none)
+        return
+    end
+    if aspect_ratio === :equal
+        aspect_ratio = 1
+    end
+    if axisletter === :y
+        attr[:aspect_ratio] = aspect_ratio*u
+        return
+    end
+    if axisletter === :x
+        attr[:aspect_ratio] = aspect_ratio/u
+        return
+    end
+    return
+end
```

Comment:
Obviously from your examples this works. But I am a bit confused how. So it seems to me that `attr[:aspect_ratio]` is typically unitful when it leaves this function. But where does that unit get stripped before it reaches Plots.jl?

Reply:
Magic.

Consider you give it an x axis in `m` and a y axis in `s`, as well as an `aspect_ratio=1u"m/s"`. First, it gets passed through the recipe for the `x` axis, where `aspect_ratio` is converted to `1u"m/s"/u"m" == 1u"s^-1"`. Then it is passed through the recipe for the `y` axis, where it is converted to `1u"s^-1"*1u"s" == 1`. If it is not unitless by now, an error has been made, and the program throws (Sadly a generic "Incompatible units" error, which might be a bit difficult to parse). 

It would have been cleaner to see some super-state where we have access to both axes' units as well as the given `aspect_ratio`, but this daisy-chain method was the only one I could see possible.

Reply:
I thought about that, but I don't know if it's guaranteed to go the right order. And if only `x` is unitful, it of course doesn't work.

Reply:
Ah right, this is slippery territory.